File: backend/app/schemas/router_event.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class RouterEventBase(BaseModel):
    event_type: str
    description: str
    event_metadata: dict | None = None


class RouterEventResponse(RouterEventBase):
    id: uuid.UUID
    router_id: uuid.UUID
    router_nombre: str | None = None
    router_ip: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/router-event.ts
export interface RouterEvent {
  id: string;
  router_id: string;
  event_type: string;
  description: string;
  event_metadata: Record<string, any> | null;
  created_at: string;
  router_nombre: string | null;
  router_ip: string | null;
}

export type RouterEventType =
  | "CREATED"
  | "UPDATED"
  | "DELETED"
  | "ONLINE"
  | "OFFLINE"
  | "IDENTITY_CHANGED"
  | "VERSION_CHANGED";
